vault/mount.go

```diff
@@ -365,6 +377,7 @@ func (c *Core) remount(src, dst string) error {
 		return err
 	}
 
+	c.mountsLock.Lock()
```

Comment:
Nope -- needs to be where it is because we need a lock anytime we touch `c.mounts`.